cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb

```diff
@@ -89,61 +89,67 @@ def self.generate_rcinfo()
+          end
+          prebuild_script = existing_prebuild_script || target.new_shell_script_build_phase("[XCRC] Prebuild")
           prebuild_script.shell_script = "\"$SCRIPT_INPUT_FILE_0\""
-          prebuild_script.input_paths = ["#{xc_location}/xcprebuild"]
+          prebuild_script.input_paths = ["$SRCROOT/#{xc_location}/xcprebuild"]
           prebuild_script.output_paths = [
-            "$(TARGET_TEMP_DIR)/rc.enabled", 
+            "$(TARGET_TEMP_DIR)/rc.enabled",
             "$(DWARF_DSYM_FOLDER_PATH)/$(DWARF_DSYM_FILE_NAME)"
           ]
           prebuild_script.dependency_file = "$(TARGET_TEMP_DIR)/prebuild.d"
 
           # Move prebuild (last element) to the first position (to make it real 'prebuild')
-          target.build_phases.rotate!(-1)
+          target.build_phases.rotate!(-1) if existing_prebuild_script.nil?
         end
 
         # Postbuild
-        postbuild_script = target.new_shell_script_build_phase("[XCRC] Postbuild")
+        existing_postbuild_script = target.build_phases.detect do |phase|
+          if phase.respond_to?(:name)
+            phase.name != nil && phase.name.start_with?("[XCRC] Postbuild")
+          end
+        end
+        postbuild_script = existing_postbuild_script || target.new_shell_script_build_phase("[XCRC] Postbuild")
         postbuild_script.shell_script = "\"$SCRIPT_INPUT_FILE_0\""
-        postbuild_script.input_paths = ["#{xc_location}/xcpostbuild"]
+        postbuild_script.input_paths = ["$SRCROOT/#{xc_location}/xcpostbuild"]
         postbuild_script.output_paths = [
-          "$(TARGET_BUILD_DIR)/$(MODULES_FOLDER_PATH)/$(PRODUCT_MODULE_NAME).swiftmodule/$(PLATFORM_PREFERRED_ARCH).swiftmodule.md5", 
+          "$(TARGET_BUILD_DIR)/$(MODULES_FOLDER_PATH)/$(PRODUCT_MODULE_NAME).swiftmodule/$(PLATFORM_PREFERRED_ARCH).swiftmodule.md5",
           "$(TARGET_BUILD_DIR)/$(MODULES_FOLDER_PATH)/$(PRODUCT_MODULE_NAME).swiftmodule/$(PLATFORM_PREFERRED_ARCH)-$(LLVM_TARGET_TRIPLE_VENDOR)-$(SWIFT_PLATFORM_TARGET_PREFIX)$(LLVM_TARGET_TRIPLE_SUFFIX).swiftmodule.md5"
         ]
         postbuild_script.dependency_file = "$(TARGET_TEMP_DIR)/postbuild.d"
 
         # Mark a sha as ready for a given platform and configuration when building the final_target
         if mode == 'producer' && target.name == final_target
-          mark_script = target.new_shell_script_build_phase("[XCRC] Mark")
+          existing_mark_script = target.build_phases.detect do |phase|
+            if phase.respond_to?(:name)
+              phase.name != nil && phase.name.start_with?("[XCRC] Mark")
+            end
+          end
+          mark_script = existing_mark_script || target.new_shell_script_build_phase("[XCRC] Mark")
           mark_script.shell_script = "\"$SCRIPT_INPUT_FILE_0\" mark --configuration $CONFIGURATION --platform $PLATFORM_NAME"
-          mark_script.input_paths = ["#{xc_location}/xcprepare"]
+          mark_script.input_paths = ["$SRCROOT/#{xc_location}/xcprepare"]
         end
```

Comment:
similarly, we need to delete `"[XCRC] Mark"` if previous one is not needed. 
Note: someone may change `final_target` and then we have to remove that step too.